src/extra/gfx/drawable.cpp

```diff
@@ -234,8 +285,70 @@ struct DrawShard : public BaseConsumer {
   SHVar activate(SHContext *shContext, const SHVar &input) { throw ActivationError("GFX.Draw: Unsupported input type"); }
 };
 
+struct DrawQueueShard {
+  static SHTypesInfo inputTypes() { return CoreInfo::AnyType; }

```

Comment:
If the input isn't used and doesn't pass through, I think we should define it as `CoreInfo::NoneType` instead of `Any`.

Reply:
Doesn't that break if you have any previously unconsumed output?
Say you do some math and then you try to use this shard, It'll complain that input is not none?

Reply:
@sinkingsugar

Reply:
UX and future UI wise setting None is actually a good point. 👍 (we won't show any connection possible on top of the shard visualization!) But I think None will spawn other issues (as it's used during compose maybe?) Can try to set None and see if works out!

Reply:
Ok let me try

Reply:
I have used `None` in other places (mostly UI) without trouble so far. I also thinks it is good to indicate that either a shard is not affected by whatever inputs comes before, or that it doesn't return anything (kind of a black hole), though I don't think we have anything right now that outputs `None` (we usually pass through).

Reply:
Yeah no issues, will annotate with NoneType for optional & no input/output from now on.